### ISSUE TITLE
Incomplete check and wrong error message

### ISSUE THREAD

**Issue by ssten** [FreeRDP/Remmina#1322]
I use remmina (=net-misc/remmina-1.2.0_rc21) with ssh and rdp plugins on Gentoo GNU/Linux (amd64, up to date system, portage tree synced at Mon Oct 16 00:39:24 UTC 2017).

$ remmina --version
(remmina:6125): Gtk-WARNING **: Theme parsing error: gtk.css:9:22: The :focused pseudo-class is deprecated. Use :focus instead.
StatusNotifier/Appindicator support: not supported by desktop. Remmina will try to fallback to GtkStatusIcon/xembed
Remmina - Version 1.2.0-rcgit-21 (git rcgit-21)

Please note Gtk-Warning.

For now I try to connect to Windows Server via rdp.

$ xfreerdp --version
This is FreeRDP version 2.0.0-dev (git n/a)

AFAIR it should be Windows Server 2012R2, for now I can't tell more details.

Remmina reports standard error message about invalid credentials:
![remmina_connection_error](https://user-images.githubusercontent.com/2209401/31759792-3b6aead4-b4bb-11e7-8933-ebe2b9987be5.png)
But really user/password are not checked. This is network settings issue.
Windows RDP client reports the following error message (excuse me for russian locale, the message means, that server requires network verification, which is not supported by the client).
![rdp_err](https://user-images.githubusercontent.com/2209401/31759807-47157b56-b4bb-11e7-8d46-05e94c04eb2e.png)

**Comment by giox069** [comment]
@ssten, can you try to change the security type in the Advanced tab of the profile settings?

![immagine](https://user-images.githubusercontent.com/5477803/31761592-c0c40ee8-b4b8-11e7-87f4-b1ef2106a80e.png)

Also please, launch remmina from a terminal with ```LANGUAGE=C remmina``, connect and reproduce the error. Copy an paste the output text here. It may contains useful informations.
Thank you.

**Comment by giox069** [comment]
Ok, thank you for testing. But I think this is a libfreerdp problem.
Can you connect with
```xfreerdp /v:servername /u:username /d:domainname``` test if it works, and paste the output here ?

**Comment by ssten** [comment]
Direct invocation of xfreerdp also fails:

> $ xfreerdp /u:user /p:password /v:host.domain.tld
> [12:42:05:338] [7060:7061] [ERROR][com.freerdp.core.nla] - SPNEGO failed with NTSTATUS: 0xC000006D
> [12:42:05:338] [7060:7061] [ERROR][com.freerdp.core] - freerdp_set_last_error Unknown error class [0xC000006D]
> [12:42:05:338] [7060:7061] [ERROR][com.freerdp.core.rdp] - rdp_recv_callback: CONNECTION_STATE_NLA - nla_recv_pdu() fail
> [12:42:05:338] [7060:7061] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
> [12:42:05:338] [7060:7061] [ERROR][com.freerdp.client.x11] - Freerdp connect error exit status 1

The Windows server is not in domain and it is not clear, how to handle this case in remmina.
In credentials domain string was «.\», what looks strange for me.

I agree, that it is primarily freerdp issue, but I expect remmina to provide more readable error messages.
Am I to open a separate issue in freerdp tracker to discover connection issue itself?

P.S. My build with my default ru_RU locale shows some strange squares (probaly instead of cyrillic letters) in error message.

**Comment by giox069** [comment]
When the windows server is not on a domain, just use the windows server name (netbios name) as domain name. In some cases also "." works.
For example if you are connecting to a sever called MYPC:
```xfreerdp /v:mypc_ip_address /u:username /d:MYPC```
The error code: the only error that remmina receives is an error number 0xC000006D, which is quite obscure for us. The poor documentation we have is [here](https://msdn.microsoft.com/en-us/library/cc704588.aspx) and as you can read is "The attempted logon is invalid. This is either due to a bad username or authentication information.".

If you google for 0xC000006D and RDP, you will find a lot of people having this error also with the windows RDP client. Have you tested it with the windows RDP client?

**Comment by ssten** [comment]
> Have you tested it with the windows RDP client?

Yes, of course.

The second window is error message from Windows XP.
Windows 7 RDP client connects, meaning display login window.
It looks like something related with RDP protocol version. *freerdp* seems to support only old ones.

**Comment by ssten** [comment]
I've succeed.

The remote server is **Windows Server 2012 R2 Datacenter 64-bit**.

Connetction succeed only with *tls* or *rdp* security types. *ext* fails, *nla* provides core dump.

The issued option on server is:

> Allow connections only from computers running Remote Desktop with Network Level Authentication (recommended)

Connection succeed only after it was disabled.

But freerdp fails to correctly fit stored password, I have to re-type it manually.

**Comment by giox069** [comment]
Thank you for the tests! I will try to reproduce it and open an issue to the FreeRDP team when I will able to get more details.
Please note: xfreerdp does not save passwords. Never. You have to type it manually or put it in the command line with /p:password. But putting password in the commandline is not secure, because your commandline can be observed by other processes and the history is saved in a cleartext file.

**Comment by ssten** [comment]
I may feel unclear about some details.
But remmina provides an option to save passwords.
I already use some rdp-connections with default security type (*Negotiation*) and stored passwords. They work fine.
But when I try to do the same with *tls* security type, Windows Server login fails.
One more strange detail: when password is interpreted as wrong, failed login resets it. But whet I'm failed to login with **/sec:tls** and right password, failed login don't affect it.

**Comment by giox069** [comment]
Yes, remmina saves its passwords, usually inside the gnome keyring, so they are protected and not stored cleartext (if gnome keyring o similar service is available).
And then yes, remmina "unsaves" the password when authentication fails, this is a wanted "feature".
Why does remmina not "unsave" the password when you use TLS? I think because when you connected with TLS the error is different: ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED 0x0002000C (please check again if this is the error).
Which is not really an indication that the password or username are invalid
The remmina code [here](https://github.com/FreeRDP/Remmina/blob/next/remmina-plugins/rdp/rdp_plugin.c#L1061) currently unsaves the password only when libfreerdp returns the two specific error codes FREERDP_ERROR_AUTHENTICATION_FAILED or STATUS_LOGON_FAILURE.
I currently have no idea if ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED error comes from libfreerdp or the RDP server itself. If it comes from libfreerdp, we can open an issue and help them testing.

**Comment by ssten** [comment]
Recenly used build of *remmina* is configured without *gnome-keyring* support (I don't want to use *systemd* and don't plan to do it).

You aren't right.
The error you quotes belongs failed connection session.

When connection succeed xfreerdp outpus is just:
> [15:22:41:851] [8603:8604] [ERROR][com.winpr.timezone] - Unable to get current timezone rule
> [15:22:41:875] [8603:8604] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
> [15:22:41:875] [8603:8604] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
> [15:22:41:988] [8603:8604] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem

But autologin with stored password is failed.

**Comment by weberhofer** [comment]
This issue seem to be related to FreeRDP/FreeRDP#4555

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @giox069 @ssten @weberhofer
